SPECS/xpmem/xpmem.spec

```diff
@@ -0,0 +1,249 @@
+Release: 1%{?dist}
+Summary: XPMEM: kernel modules
+Group: System Environment/Libraries
+%description modules
+XPMEM is a Linux kernel module that enables a process to map the
+memory of another process into its virtual address space. Source code
+can be obtained by cloning the Git repository, original Mercurial
+repository or by downloading a tarball from the link above.
+
+This package includes the kernel module (non KMP version).
+%endif #end if "%{KMP}" == "1"
+
+#
+# setup module sign scripts if paths to the keys are given
+#
+%global WITH_MOD_SIGN %(if ( test -f "$MODULE_SIGN_PRIV_KEY" && test -f "$MODULE_SIGN_PUB_KEY" ); \
+	then \
+		echo -n '1'; \
+	else \
+		echo -n '0'; fi)
+
+%if "%{WITH_MOD_SIGN}" == "1"
+# call module sign script
+%global __modsign_install_post \
+    $RPM_BUILD_DIR/xpmem-%{version}/tools/sign-modules $RPM_BUILD_ROOT/lib/modules/ %{kernel_source default} || exit 1 \
+%{nil}
+
+# Disgusting hack alert! We need to ensure we sign modules *after* all
+# invocations of strip occur, which is in __debug_install_post if
+# find-debuginfo.sh runs, and __os_install_post if not.
+#
+%define __spec_install_post \
+  %{?__debug_package:%{__debug_install_post}} \
+  %{__arch_install_post} \
+  %{__os_install_post} \
+  %{__modsign_install_post} \
+%{nil}
+
+%endif # end of setup module sign scripts
+#
+
+%if 0%{?rhel} > 0 || 0%{?euleros} >= 2
+%global install_mod_dir extra/%{_name}
+%endif
+
+%{!?install_mod_dir: %global install_mod_dir updates}
+
+%global moduledir /lib/modules/%{KVERSION}/%{install_mod_dir}
+
+%prep
+%setup -q
+
+%build
+env=
+if [ "$CROSS_COMPILE" != '' ]; then
+  env="$env CC=${CROSS_COMPILE}gcc"
+fi
+./autogen.sh
+%{configure} \
+  --with-module-prefix= \
+  --with-kerneldir=%{K_SRC} \
+  $env \
+  #
+%{make_build} %{make_kernel_only}
+
+%install
+%{make_install} moduledir=%{moduledir} %{make_kernel_only}
+rm -rf $RPM_BUILD_ROOT/%{_libdir}/libxpmem.la
+rm -rf $RPM_BUILD_ROOT/etc/init.d/xpmem
+mkdir -p $RPM_BUILD_ROOT%{_prefix}/lib/modules-load.d
+echo "xpmem" >$RPM_BUILD_ROOT%{_prefix}/lib/modules-load.d/xpmem.conf
+%if %{with kernel_only}
+rm -f $RPM_BUILD_ROOT/usr/lib*/pkgconfig/cray-xpmem.pc
+%endif
+%if %{need_firmware_dir}
+mkdir -p $RPM_BUILD_ROOT/lib/firmware
+%endif
+
+%clean
+rm -rf $RPM_BUILD_ROOT
+
+%if ! %{with kernel_only}
+%post   -n libxpmem -p /sbin/ldconfig
+%postun -n libxpmem -p /sbin/ldconfig
+%endif
+
+%postun
+if [ "$1" = 0 ]; then
+	if lsmod | grep -qw xpmem; then
+		# If the module fails to unload, give an error,
+		# but don't fail uninstall. User should handle this
+		# Maybe the module is in use
+		rmmod xpmem || :
+	fi
+fi
+
+%files
+/lib/udev/rules.d/*-xpmem.rules
+%{_prefix}/lib/modules-load.d/xpmem.conf
+%doc README AUTHORS
+%license COPYING COPYING.LESSER
+
+%if ! %{with kernel_only}
+%files -n libxpmem
+%{_libdir}/libxpmem.so.*
+
+%files -n libxpmem-devel
+%{_prefix}/include/xpmem.h
+%{_libdir}/libxpmem.a
+%{_libdir}/libxpmem.so
+%{_libdir}/pkgconfig/cray-xpmem.pc
```

Comment:
These two subpackages would need `%license` as well. It's because they can be installed separately from the default `xpmem` package (e.g. they don't `Require: %{name} = %{version}-%{release}`), so it means that it's possible to install a package from this spec without a license file.

To clarify: it is completely fine for a subpackage to not require the default package inside one spec. I'm only making sure we end up with license info regardless of which of the packages gets installed.

Please also verify if the subpackages from other specs don't have a similar problem.

Reply:
fixed and pushed changes.